### PR TITLE
Update angular directive for nimble-rich-text-viewer

### PR DESCRIPTION
**Pull request by vivinkrishna-ni**
# Pull Request

## 🤨 Rationale

This PR enables exposing the `markdown` property of `nimble-rich-text-viewer` to the angular application.

## 👩‍💻 Implementation

- Updates angular directive to add support for `markdown`.
- Updates example client app with a sample markdown string.

Please note that this PR is subject to change based on the other PRs for `nimble-rich-text-viewer` and is currently in progress.

## 🧪 Testing

- Added unit test for the directive.
- Tested the binding of `markdown` manually in the example client app.

## ✅ Checklist

- [x] I have updated the project documentation to reflect my changes or determined no changes are needed.

### CODE REVIEW

**Review by vikisekarNI — APPROVED** [review]

File: angular-workspace/projects/ni/nimble-angular/rich-text-viewer/tests/nimble-rich-text-viewer.directive.spec.ts [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+            expect(directive.markdown).toBe('');
+            expect(nativeElement.markdown).toBe('');
+        });
+    });
+
+    describe('with template string values', () => {
+        @Component({
+            template: `
+                <nimble-rich-text-viewer #viewer
+                    markdown="Markdown value"
```
[the remark this answers is not in the document]
Comment (vivinkrishna-ni) [reply angular-workspace/projects/ni/nimble-angular/rich-text-viewer/tests/nimble-rich-text-viewer.directive.spec.ts]: Updated as suggested.

File: change/@ni-nimble-angular-1bea987a-0009-4367-bf30-2b0921ae26c2.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (jattasNI) [inline change/@ni-nimble-angular-1bea987a-0009-4367-bf30-2b0921ae26c2.json:1]: Placeholder comment to remind you to:
1. retarget this PR to `main` once the predecessor PR goes in
2. ensure there are no Chromatic diffs reported after that build completes (I assume the diffs in the current build are from the predecessor PR)

File: angular-workspace/projects/example-client-app/src/app/customapp/customapp.component.scss [hunk 3]
```diff
@@ -77,3 +77,9 @@ nimble-table {
 .query-param-word {
     margin-top: $ni-nimble-standard-padding;
 }
+
+.rich-text-viewer-container {
```
Comment (jattasNI) [inline angular-workspace/projects/example-client-app/src/app/customapp/customapp.component.scss:81]: Is this container always going to be necessary or will it go away when we adopt new styling that more clearly separates the viewer from its background? If it will go away could you make sure the work to delete it is tracked in a work item?

File: angular-workspace/projects/ni/nimble-angular/rich-text-viewer/tests/nimble-rich-text-viewer.directive.spec.ts [hunk 4]
```diff
@@ -13,4 +15,111 @@ describe('Nimble Rich Text Viewer', () => {
...
+            expect(directive.markdown).toBeUndefined();
+            expect(nativeElement.markdown).toBeUndefined();
+        });
+    });
+
+    describe('with template string values', () => {
+        @Component({
+            template: `
+                <nimble-rich-text-viewer #viewer
+                    markdown="**Bold** _Italics_ \n\n1. \n2. \n * \n * \n"
```
Comment (jattasNI) [inline angular-workspace/projects/ni/nimble-angular/rich-text-viewer/tests/nimble-rich-text-viewer.directive.spec.ts:55]: I'm curious for other reviewers' opinions about whether we should add test coverage for various string values (e.g. values that need escaping) or if we can rely on Angular to do conversions for us.

**Review by jattasNI — APPROVED** [review]

File: angular-workspace/projects/example-client-app/src/app/customapp/customapp.component.scss (hunk 3 above)
Comment (vivinkrishna-ni) [reply angular-workspace/projects/example-client-app/src/app/customapp/customapp.component.scss:81]: I added this container to make the component separated from the background as you mentioned and it is only for the example client app representation. As per the [latest visuals for the nimble viewer component](https://www.figma.com/file/7tTImJ8UueTcmgqcCWmD5q/Rich-Text-Editor-Visuals?node-id=7565%3A31340&mode=dev), the appearance will be transparent and does not have any borders. It is completely based on the client's needs whether to set the border or not.

Comment (jattasNI) [reply angular-workspace/projects/example-client-app/src/app/customapp/customapp.component.scss:81]: Actually, looking at your Figma link more closely I'm not seeing what you mean about the component not having borders. At least visually it looks like they do. Has something changed or am I missing something?

image

Comment (vivinkrishna-ni) [reply angular-workspace/projects/example-client-app/src/app/customapp/customapp.component.scss:81]: I meant that the placeholder that is marked inside the red border in the image below, is the specific component I am referring to as `nimble-rich-text-viewer`. Other details such as the username and timestamp will be included as part of the `sl-comment` component ([sl-comment-module HLD](https://dev.azure.com/ni/DevCentral/_git/Skyline?path=/docs/design-documents/Platform/Comments/Sl-Comments-Module-Design.md&version=GBusers/susee/comments-hld/sl-comments-module&line=140&lineEnd=141&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents) - currently in PR state.), and the border will encompass both the header (username + timestamp) and the `nimble-rich-text-viewer`.
image

Our idea was to make the nimble viewer component borderless by default for now, allowing the client to choose whether to include borders. Later when we incorporate a setting like `appearance`, we will update it accordingly.

Comment (jattasNI) [reply angular-workspace/projects/example-client-app/src/app/customapp/customapp.component.scss:81]: Ah, right, I'd forgotten about where the boundary was. Given that's our current approved direction then I think what you're doing here makes sense. As we get more clients we should keep an eye on whether the username and timestamp should move into nimble, but that's not a decision we need to revisit now.

File: angular-workspace/projects/example-client-app/src/app/customapp/customapp.component.ts [hunk 5]
```diff
@@ -41,6 +41,7 @@ export class CustomAppComponent {
     public selectedRadio = 'mango';
     public activeTabId = 'tab-1';
     public activeAnchorTabId = 'a-tab-2';
+    public markdownString = 'Supported rich text formatting options:\n1. **Bold**\n2. *Italics*\n3. Numbered lists\n   1. Option 1\n   2. Option 2\n4. Bulleted lists\n   * Option 1\n   * Option 2\n5. Direct link: <https://nimble.ni.dev/>';
```
Comment (rajsite) [inline angular-workspace/projects/example-client-app/src/app/customapp/customapp.component.ts:44]: You can use template string literals to make this look nicer andd easier to edit as markdown:

```suggestion
    public markdownString = `Supported rich text formatting options:
1. **Bold**
2. *Italics*
3. Numbered lists
   1. Option 1
   2. Option 2
4. Bulleted lists
   * Option 1
   * Option 2
5. Direct link: <https://nimble.ni.dev/>
`;
```

File: angular-workspace/projects/ni/nimble-angular/rich-text-viewer/tests/nimble-rich-text-viewer.directive.spec.ts (hunk 4 above)
Comment (rajsite) [reply angular-workspace/projects/ni/nimble-angular/rich-text-viewer/tests/nimble-rich-text-viewer.directive.spec.ts:55]: I'm betting the template string values are mostly an edge case and at most have simple initial text. I expect the property binding to be more common and unlikely to cause issues as the value is passed straight through.

I'm okay not being overly robust testing angular's template string behavior. It is whatever angular has chosen and I suspect is documented somewhere. But if folks have issues with it I suspect switching to a binding would be what they try anyway. At the same time not against the tests if wanted :P

**Review by rajsite — APPROVED** [review]

File: angular-workspace/projects/example-client-app/src/app/customapp/customapp.component.scss (hunk 3 above)
Comment (rajsite) [reply angular-workspace/projects/example-client-app/src/app/customapp/customapp.component.scss:81]: I don't think we would add specific name and date slots to a generic rich text viewer / editor, but I could imagine requests in the future to have the viewer have an appearance configuration so it looks and behaves more like a form control (like a text area) instead of a block of text (like a p tag). But yea something we can keep an eye on and address in the future.

File: angular-workspace/projects/example-client-app/src/app/customapp/customapp.component.ts (hunk 5 above)
Comment (vivinkrishna-ni) [reply angular-workspace/projects/example-client-app/src/app/customapp/customapp.component.ts]: Changed to template string literals

File: angular-workspace/projects/example-client-app/src/app/customapp/customapp.component.scss (hunk 3 above)
Comment (vivinkrishna-ni) [reply angular-workspace/projects/example-client-app/src/app/customapp/customapp.component.scss:81]: Sure, I will resolve this comment for now and will address this in the future.

File: angular-workspace/projects/ni/nimble-angular/rich-text-viewer/tests/nimble-rich-text-viewer.directive.spec.ts (hunk 4 above)
Comment (vivinkrishna-ni) [reply angular-workspace/projects/ni/nimble-angular/rich-text-viewer/tests/nimble-rich-text-viewer.directive.spec.ts:55]: I have added a test case with the template string literal in the property binding case.

**Review by atmgrifter00 — APPROVED** [review]

File: angular-workspace/projects/ni/nimble-angular/rich-text-viewer/tests/nimble-rich-text-viewer.directive.spec.ts [hunk 6]
```diff
@@ -13,4 +15,128 @@ describe('Nimble Rich Text Viewer', () => {
...
+
+            expect(directive.markdown).toBe('***new markdown value***');
+            expect(nativeElement.markdown).toBe('***new markdown value***');
+        });
+
+        it('can be configured with property binding for markdown in a string literals', () => {
+            expect(directive.markdown).toBe('**Bold** _Italics_ \n\n1. \n2. \n * \n * \n');
+            expect(nativeElement.markdown).toBe('**Bold** _Italics_ \n\n1. \n2. \n * \n * \n');
+
+            fixture.componentInstance.markdown = `New value
```
[the remark this answers is not in the document]
Comment (vivinkrishna-ni) [reply angular-workspace/projects/ni/nimble-angular/rich-text-viewer/tests/nimble-rich-text-viewer.directive.spec.ts]: Updated as suggested.

File: change/@ni-nimble-angular-1bea987a-0009-4367-bf30-2b0921ae26c2.json (hunk 2 above)
Comment (vivinkrishna-ni) [reply change/@ni-nimble-angular-1bea987a-0009-4367-bf30-2b0921ae26c2.json:1]: I have verified the Chromatic differences once retargeted this PR to `main` and there are no differences. So I am closing this comment.
